src/WaveTrack.cpp

```diff
@@ -64,6 +64,37 @@ from the project that will own the track.
 
 using std::max;
 
+namespace {
+
+bool AreAligned(const WaveClipHolders& a, const WaveClipHolders& b)
+{
+    if (a.size() != b.size())
+        return false;
+
+    const auto less = [](const WaveClip* a, const WaveClip* b) {
+        return a->GetOffset() < b->GetOffset();
+    };
+
+    std::vector<WaveClip*> sorted;
+    sorted.reserve(a.size());
+    for (const auto& clip : b)
+    {
+        auto at = std::lower_bound(sorted.begin(), sorted.end(), clip.get(), less);
+        sorted.insert(at, clip.get());
+    }
+    for (const auto& clip : a)
+    {
+        auto it = std::lower_bound(sorted.begin(), sorted.end(), clip.get(), less);
+        if (it == sorted.end()
+            || clip->GetOffset() != (*it)->GetOffset()
+            || clip->GetNumSamples() != (*it)->GetNumSamples())
+            return false;
+    }
+    return true;
+}
```

Comment:
And then instead, you might use std::mismatch on the two ranges of pointers, and the appropriate lambda, expressing what you mean very simply.